pkg/controllers/user/logging/projectlogging.go

```diff
@@ -0,0 +1,107 @@
+package logging
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/types/apis/apps/v1beta2"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	rbacv1 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	"github.com/rancher/types/config"
+	"k8s.io/apimachinery/pkg/labels"
+
+	loggingconfig "github.com/rancher/rancher/pkg/controllers/user/logging/config"
+	"github.com/rancher/rancher/pkg/controllers/user/logging/generator"
+	"github.com/rancher/rancher/pkg/controllers/user/logging/utils"
+)
+
+const (
+	ProjectIDAnnotation = "field.cattle.io/projectId"
+)
+
+type ProjectLoggingSyncer struct {
+	management           config.ManagementContext
+	projectLoggings      v3.ProjectLoggingInterface
+	clusterLoggingLister v3.ClusterLoggingLister
+	namespaceLister      v1.NamespaceLister
+	serviceAccounts      v1.ServiceAccountInterface
+	configmaps           v1.ConfigMapInterface
+	daemonsets           v1beta2.DaemonSetInterface
+	clusterRoleBindings  rbacv1.ClusterRoleBindingInterface
+}
+
+func registerProjectLogging(cluster *config.UserContext) {
+	projectLoggings := cluster.Management.Management.ProjectLoggings("")
+	syncer := &ProjectLoggingSyncer{
+		management:           *cluster.Management,
+		projectLoggings:      projectLoggings,
+		clusterLoggingLister: cluster.Management.Management.ClusterLoggings("").Controller().Lister(),
+		serviceAccounts:      cluster.Core.ServiceAccounts(loggingconfig.LoggingNamespace),
+		namespaceLister:      cluster.Core.Namespaces("").Controller().Lister(),
+		configmaps:           cluster.Core.ConfigMaps(loggingconfig.LoggingNamespace),
+		daemonsets:           cluster.Apps.DaemonSets(loggingconfig.LoggingNamespace),
+		clusterRoleBindings:  cluster.RBAC.ClusterRoleBindings(loggingconfig.LoggingNamespace),
+	}
+	fmt.Println("-----------before")
+	projectLoggings.AddClusterScopedHandler("project-logging-controller", cluster.ClusterName, syncer.Sync)
+	fmt.Println("-----------after")
+}
+
+func (c *ProjectLoggingSyncer) Sync(key string, obj *v3.ProjectLogging) error {
+	//clean up
+	if obj == nil || obj.DeletionTimestamp != nil {
+		allDisabled, err := utils.IsAllLoggingDisable(c.clusterLoggingLister, c.projectLoggings.Controller().Lister())
+		if err != nil {
+			return err
+		}
+
+		if allDisabled {
+			if err := utils.RemoveFluentd(c.daemonsets, c.serviceAccounts, c.clusterRoleBindings); err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := c.createOrUpdateProjectConfigMap(); err != nil {
```

Comment:
Shouldn't we return when object == nil || obj.DeletionTimestamp != nil? Looks like line 67 will still be hit